lib/theory/src/Theory/Tools/Wellformedness.hs

```diff
@@ -457,6 +476,64 @@ factReports thy = concat
         : (do RuleItem ru <- get thyItems thy; get rActs $ get oprRuleE ru)
           ++ (do RuleItem ru <- get thyItems thy; racs <- get oprRuleAC ru; get rActs racs)
 
+    
+    -- | To bound a fact with his most similar fact
+    -- | The editing distance of two facts must between 1(includ) and 3 (includ)
+    mostSimilarName :: [RuleAndFact]->[RuleAndFact]
+                      ->[(RuleAndFact,RuleAndFact)]
+    mostSimilarName xs xt = 
+        filter (\x-> (notSame (fst x) $ snd x)
+         && (isSimiler (fst x) $ snd x) ) $ 
+        foldr (\x acc-> ((,) x $ minEd (snd x) xt):acc) [] xs
+      where
+        isSimiler s t             = distance (snd s) t < 3
+        notSame s t               = distance (snd s) t/=0 
+        minEd s li                = head $
+                                    sortOn (\x -> distance s x) li
+        distance factL ruleRactR  = editDistance (getName factL) $ 
+                                    getName $ snd ruleRactR
+        tagName (tag,_,_)         = factTagName tag
+        getName fact              = tagName $ factInfo fact
+    
+    
+    -- Report a protocol fact occurs in an LHS but not in any RHS
+    factLhsOccurNoRhs = 
+      case factLhsNoRhs of
+        []            -> []
+        facts         -> return $ (,) topic $ numbered' $
+                          map (nest 2 . ruleAndFact ) facts
+      where
+        topic = "Facts occur in an left-hand-side but not in any right-hand-side "
+        factLhsNoRhs = [fa | fa <-factsLhsNoRhs 
```

Comment:
factLhsNoRhs vs factsLhsNoRhs ? Please choose better names.

Reply:
Resolved, the new name for "factsLhsNoRhs" is "getFactLhsNoRhs"